Automattic/wp-calypso
# format-currency: Set currency locale when changing translation locale

#### Proposed Changes

When changing the translation language of calypso, this PR adds a secondary side effect which also sets the locale used by the `formatCurrency()` function.

This side effect is added to the `setLocale()` redux action creator for the UI state, which is awkward because action creators should not have side effects. However, this places it alongside the other side effect (which sets the translation locale) which has been in this location since https://github.com/Automattic/wp-calypso/pull/18297... although reading https://github.com/Automattic/wp-calypso/pull/58634 I wonder if maybe I should put this someplace else? cc @jsnajdr 

Before (with en browser locale, WordPress.com user interface set to Canadian French, and currency set to CAD):

<img width="575" alt="before-setting-currency-locale" src="https://user-images.githubusercontent.com/2036909/211662493-fd46bc5f-eb5a-49b5-92f6-8f977c176a33.png">

After (with en browser locale, WordPress.com user interface set to Canadian French, and currency set to CAD):

<img width="568" alt="after-setting-currency-locale" src="https://user-images.githubusercontent.com/2036909/211662510-17097e04-774e-4024-8e71-9d451ef7b616.png">

Fixes https://github.com/Automattic/wp-calypso/issues/71816

#### Testing Instructions

- Use a web browser with its locale set to `en`. (If you need to change your web browser's locale, you can use the devtool "Sensors" panel in Google Chrome.)
- Set your user's currency to CAD.
- Set your user's user interface language to "Canadian French" using the form at `/me/account`.

<img width="502" alt="Screenshot 2023-01-10 at 4 07 48 PM" src="https://user-images.githubusercontent.com/2036909/211662775-d3571a12-6c3e-49b8-803c-45eb605fe673.png">

- Now add a product to your cart and visit checkout.
- Compare checkout with or without this PR. Verify that without this PR, the price is rendered with the currency symbol (eg: `C$123` which is the `en` formatting) before the number and that after this PR the currency symbol is rendered after the number (eg: `123 C$` which is the `fr-CA` formatting).
